code/modules/mob/living/carbon/monkey/combat.dm

```diff
@@ -0,0 +1,436 @@
+			for(var/mob/living/L in around)
+				if(L != target && enemies[L] && L.stat == CONSCIOUS && prob(25))
+					target = L
+					return TRUE
+
+			// if can't reach target for long enough, go idle
+			if(frustration >= 8)
+				back_to_idle()
+				return TRUE
+
+			if(target && target.stat == CONSCIOUS)		// make sure target exists
+				if(Adjacent(target) && isturf(target.loc))	// if right next to perp
+					var/obj/item/weapon/Weapon = locate(/obj/item/weapon) in held_items
+
+					// attack with weapon if we have one
+					if(Weapon)
+
+						// if the target has a weapon, 50% change to disarm them
+						if((locate(/obj/item/weapon) in target.held_items) && prob(50))
+
+							pickupTarget = locate(/obj/item/weapon) in target.held_items
+
+							a_intent = INTENT_DISARM
+							target.attackby(Weapon, src)
+							attacked(target)
+
+						else
+							a_intent = INTENT_HARM
+							target.attackby(Weapon, src)
+							attacked(target)
+					else
+
+						// if the target has a weapon, 50% change to disarm them
+						if((locate(/obj/item/weapon) in target.held_items) && prob(50))
+
+							pickupTarget = locate(/obj/item/weapon) in target.held_items
+
+							a_intent = INTENT_DISARM
+							target.attack_paw(src)
+							attacked(target)
+
+						else
+							a_intent = INTENT_HARM
+							target.attack_paw(src)
+							attacked(target)
+					return TRUE
+
+				else								// not next to perp
+					var/turf/olddist = get_dist(src, target)
+					if((get_dist(src, target)) >= (olddist))
+						frustration++
+					else
+						frustration = 0
+			else
+				back_to_idle()
+
+		if(MONKEY_FLEE)
+			var/list/around = view(src, vision_range)
+			target = null
+			for(var/mob/living/carbon/C in around)
+				if(enemies[C] && C.stat == CONSCIOUS)
+					target = C
+
+			if(target != null)
+				walk_away(src, target, vision_range, 5)
+			else
+				mode = MONKEY_IDLE
+
+			return TRUE
+
+		if(MONKEY_DISPOSE)
+
+			// if can't dispose of body go back to idle
+			if(!target || !bodyDisposal || frustration >= 16)
+				back_to_idle()
+				return TRUE
+
+			if(target.pulledby != src)
+
+				walk2derpless(target.loc)
+
+				if(Adjacent(target) && isturf(target.loc))
+					a_intent = INTENT_GRAB
+					target.grabbedby(src)
+				else
+					var/turf/olddist = get_dist(src, target)
+					if((get_dist(src, target)) >= (olddist))
+						frustration++
+					else
+						frustration = 0
+
+			else
+				walk2derpless(bodyDisposal.loc)
+
+				if(Adjacent(bodyDisposal))
+					bodyDisposal.stuff_mob_in(target, src)
+				else
+					var/turf/olddist = get_dist(src, bodyDisposal)
+					if((get_dist(src, bodyDisposal)) >= (olddist))
+						frustration++
+					else
+						frustration = 0
+
+			return TRUE
+
+
+
+	return resisting
+
+/mob/living/carbon/monkey/proc/back_to_idle()
+	mode = MONKEY_IDLE
+	target = null
+	a_intent = INTENT_HELP
+	frustration = 0
+
+// handle de-aggro
+/mob/living/carbon/monkey/proc/attacked(mob/living/carbon/H)
+	if(prob(25))
+		enemies[H] --
```

Comment:
Incrementing uninitilised member of a list sets it to 1. I'm using the list as a Map. Mapping from Mob to Aggression. If a mob is in the map it is an enemy. The number is used in the de-aggro mechanic.